tests/s390x_tests/TOOL_s390_chchp/chchpmain.sh

```diff
@@ -0,0 +1,199 @@
+# Copyright (C) 2018 IBM Corp.
+# 
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+
+#!/bin/bash
+
+# Test case Description: the main goal of this test is to verify the
+# basic functionality of chchp and lschp tools. In order to do so,
+# initially basic options of both commands are tested (help menu and
+# tool version). Then, using chchp, a channel path given as parameter
+# is disabled, enabled, deconfigured and configured. lschp is used to
+# check if the channel path current state is the expected one.
+# This script if the basic options of chchp are working properly and
+# disables, enables, deconfigures and configures a channel path using
+# chchp. Then verifies the result with lschp.
+
+source "lib/auxx.sh" || exit 1
+source "lib/env.sh" || exit 1
+
+chp::getVary() {
+	CHP="$1";
+	CSS="0";
+	test -z "${CHP}" && return 1;
+	if [[ "${CHP}" =~ [:digit:]*\.[:digit:]* ]]; then
+		CSS="${CHP%.*}";
+		CSS="${CSS:=0}";
+		CHP="${CHP#*.}";
+	fi
+
+	if [ "$(cat "/sys/devices/css${CSS}/chp${CSS}.${CHP}/status")" == "online" ]; then
+		echo 1;
+	elif [ "$(cat "/sys/devices/css${CSS}/chp${CSS}.${CHP}/status")" == "offline" ]; then
+		echo 0;
+	else
+		return 1;
+	fi
+	return 0;
+}
+
+chp::getConfigure() {
+	CHP="$1";
+	CSS="0";
+	test -z "${CHP}" && return 1;
+	if [[ "${CHP}" =~ [:digit:]*\.[:digit:]* ]]; then
+		CSS="${CHP%.*}";
+		CSS="${CSS:=0}";
+		CHP="${CHP#*.}";
+	fi
+
+	cat "/sys/devices/css${CSS}/chp${CSS}.${CHP}/configure";
+	return 0;
+}
+
+verifyBasicOptions() {
+	assert_exec 0 "chchp --help"
+	assert_exec 0 "chchp -h"
+	assert_exec 0 "chchp --version"
+
+	if chchp --help | grep -q "[[:space:]]*-V,"; then
+		echo "Found -V in --help information";
+		assert_exec 0 "chchp -V";
+	else
+		echo "Not found -V in --help information - skipping test";
+	fi
+
+	echo "Verify that -v is still vary and not version";
+	chchp -v 2>&1 | grep -Eq "(Option '-v' requires an argument|chchp: --vary requires an argument)";
+	[[ "${PIPESTATUS[0]}" == 1 && "${PIPESTATUS[1]}" == 0 ]];
+	assert_warn $? 0 "Verify 'chchp -v' still requires an argument";
+}
+
+verifyVaryAndConfigure() {
+	# Use --vary and -v
+	echo "Turn off CHPID $chpid_1 using Vary"
+	assert_exec 0 "chchp --vary 0 $chpid_1"
+	if [[ "$(chp::getVary "${chpid_1}")" -eq 0 ]]; then
+		echo "CHPID $chpid_1 vary OFF was successful"
+	else
+		echo "CHPID vary OFF failed"
+	fi
+
+	sleep 5
```

Comment:
"magic number" - better to extract to a variable

Reply:
Ok, changed that to a variable